src/components/ConfirmedRoute.js

```diff
@@ -0,0 +1,118 @@
+import ONYXKEYS from '../ONYXKEYS';
+import CONST from '../CONST';
+import * as MapboxToken from '../libs/actions/MapboxToken';
+import * as Expensicons from './Icon/Expensicons';
+import theme from '../styles/themes/default';
+import styles from '../styles/styles';
+import transactionPropTypes from './transactionPropTypes';
+import BlockingView from './BlockingViews/BlockingView';
+import useNetwork from '../hooks/useNetwork';
+import useLocalize from '../hooks/useLocalize';
+
+const propTypes = {
+    /** Transaction that stores the distance request data */
+    transaction: transactionPropTypes,
+
+    /** Token needed to render the map */
+    mapboxToken: PropTypes.string,
+};
+
+const defaultProps = {
+    transaction: {},
+    mapboxToken: '',
+};
+
+const getWaypointMarkers = (waypoints) => {
+    const numberOfWaypoints = _.size(waypoints);
+    const lastWaypointIndex = numberOfWaypoints - 1;
+    return _.filter(
+        _.map(waypoints, (waypoint, key) => {
+            if (!waypoint || waypoint.lng === undefined || waypoint.lat === undefined) {
+                return;
+            }
+
+            const index = Number(key.replace('waypoint', ''));
+            let MarkerComponent;
+            if (index === 0) {
+                MarkerComponent = Expensicons.DotIndicatorUnfilled;
+            } else if (index === lastWaypointIndex) {
+                MarkerComponent = Expensicons.Location;
+            } else {
+                MarkerComponent = Expensicons.DotIndicator;
+            }
+
+            return {
+                coordinate: [waypoint.lng, waypoint.lat],
+                markerComponent: () => (
+                    <MarkerComponent
+                        width={CONST.MAP_MARKER_SIZE}
+                        height={CONST.MAP_MARKER_SIZE}
+                        fill={theme.icon}
+                    />
+                ),
+            };
+        }),
+        (waypoint) => waypoint,
+    );
+};
+
+function ConfirmedRoute({mapboxToken, transaction}) {
+    const {isOffline} = useNetwork();
+    const {translate} = useLocalize();
+    const {route0: route} = transaction.routes || {};
+    const waypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const coordinates = lodashGet(route, 'geometry.coordinates', []);
+    const waypointMarkers = getWaypointMarkers(waypoints);
+
+    useEffect(() => {
+        MapboxToken.init();
+        return MapboxToken.stop;
+    }, []);
+
+    return (
+        <>
+            {!isOffline && mapboxToken ? (
+                <MapView
+                    accessToken={mapboxToken}
+                    mapPadding={CONST.MAP_PADDING}
+                    pitchEnabled={false}
+                    directionCoordinates={coordinates}
+                    directionStyle={styles.mapDirection}
+                    style={styles.mapView}
+                    waypoints={waypointMarkers}
+                    styleURL={CONST.MAPBOX_STYLE_URL}
+                />
```

Comment:
We could pass `initialState` prop with start waypoint to avoid locating wrong location on first render.
Issue: https://github.com/Expensify/App/issues/26726